iface-ng/iface-ng

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash
+# Written by Roberto Santalla <roobre@roobre.es>
+# Based on iface-ng, Copyright (C) 2014 Julien Bonjean <julien@bonjean.info> and Alexander Keller <github@nycroth.com>
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#------------------------------------------------------------------------
+
+usage() {
+    echo "Usage: $0 [-d down] [-n \"no ip\"] [-c FF0000] [-C 00FF00] [-e] [-p] [-s] [-6]
+    -d down: Placeholder text if the specified interface is down. If set to an empty string,
+       the entire block is hidden (useful to save space on small screens).
+    -n \"no ip\": Placeholder text to display when the interface is up but it does not
+       have any ip address.
+    -c FF0000: Text color when interface is down.
+    -C 00FF00: Text color when interface is up.
+    -e: Show interface even if it does not exist (useful for tun/tap).
+    -p: Include CIDR prefix.
+    -s: Include link speed (ethernet only).
+    -6: Use ipv6 address insead of v4." 1>&2
+    exit 1; 
+}
+
+# Default options
+downText="down"
+noipText="no ip"
+colorDown="FF0000"
+colorUp="00FF00"
+inet="inet"
+e=0
+p=0
+s=0
+
+while getopts "d:n:c:C:esp6" opt; do
+    case "${opt}" in
+        d)
+            downText=${OPTARG}
+            ;;
```

Comment:
I don't think there is a need for each of these clauses to take up three lines since there isn't much logic going on. In this case I would recommend the style

    d) downText=${OPTARG} ;;
    n) noipText=${OPTARG} ;;
    ...